chosen.js

```diff
@@ -90,6 +90,27 @@
           if (attr.ngOptions && ngModel) {
             match = attr.ngOptions.match(NG_OPTIONS_REGEXP);
             valuesExpr = match[7];
+            var options = scope.$eval(valuesExpr);
+            // Adding title in case of array of options and object data source
+            if (options && typeof options === 'object') {
+                /*Timeout is required to make it work. Without timeout, options with title are formed initially but because of ng-options, some how
+                options are rendered again and removed all titles from options
+                TODO: will debug it to find out reason */
+                $timeout(function() {
+                    // Convert object into array first and then assign titles to all options according to values of object
+                    if(!(options instanceof Array)) {
+                        options = Object.keys(options).map(function(key) {
+                            return options[key]
+                        })
+                    }
+                    for (var index=options.length - 1,lastOptionIndex =  $(element).find("option").length - 1;index>= 0;index--,lastOptionIndex--){
```

Comment:
Can we store the element `$(element).find("option").length` in a variable as its used below too.

Reply:
done